api/openapi.yaml

```diff
@@ -3078,66 +3078,90 @@ components:
+    NotificationEventCommonPayload:
+      description: Common fields for notification event payload.
       type: object
       required:
         - id
         - type
         - timestamp
-        - data
       properties:
         id:
           description: A unique identifier for the notification event the payload belongs to.
           type: string
           example: "01J2KNP1YTXQRXHTDJ4KPR7PZ0"
         type:
-          description: The type of the notification rule generated this event.
-          type: string
-          example: "entitlements.balance.threshold"
-        metadata:
-          description: Metadata information related to the event.
-          type: object
-          additionalProperties: true
-          nullable: true
+          $ref: "#/components/schemas/NotificationRuleType"
         timestamp:
           description: Timestamp when the notification event was created.
           type: string
           format: date-time
           example: "2023-01-01T00:00:00Z"
-        data:
-          description: |
-            Event type specific data.
 
-            The format may vary based on the type of the event and the schema is defined for each event type separately.
-          type: object
-          additionalProperties: true
-          example:
-            currentBalance: 10000
-            threshold: "120%"
-            feature:
-              archivedAt: "2023-01-01T00:00:00Z"
-              key: "gpt4_tokens"
-              name: "AI Tokens"
-              metadata:
-                additionalProp1: "value1"
-              meterSlug: "tokens_total"
-              meterGroupByFilters:
-                model: "gpt-4"
-              id: "01ARZ3NDEKTSV4RRFFQ69G5FAV"
-              createdAt: "2023-01-01T00:00:00Z"
-              updatedAt: "2023-01-01T00:00:00Z"
-            subject:
-              id: "01G65Z755AFWAKHE12NY0CQ9FH"
-              key: "customer-id"
-              displayName: "Customer Name"
-              metadata:
-                hubspotId: "123456"
-              currentPeriodStart: "2023-01-01T00:00:00Z"
-              currentPeriodEnd: "2023-02-01T00:00:00Z"
+    NotificationEventBalanceThresholdPayload:
+      description: |
+        Defines payload specific for notification event with type
+      allOf:
+        - $ref: "#/components/schemas/NotificationEventCommonPayload"
+        - type: object
+          required:
+            - data
+          properties:
+            data:
+              type: object
+              required:
+                - entitlement
+                - feature
+                - subject
+                - balance
+              properties:
+                entitlement:
+                  $ref: "#/components/schemas/EntitlementMetered"
+                feature:
+                  $ref: "#/components/schemas/Feature"
+                subject:
+                  $ref: "#/components/schemas/Subject"
+                balance:
```

Comment:
Isn't there a balance type?

Reply:
So I was kinda hasitant to use `EntitlementValue` directly as it is shared across all the entitlement types meaning it has fields for all teh entitlement types while we only need `metered` here. So it might be confusing for users to see optinal fields in the spec which will be never used for `entitlements.balance.threshold` payload.

It would be awesome to have entitlement specific `Entitlement[Metered|Static|Boolean]Value` resources in the spec to support better reusability, but making this type of change out of scope for this PR.

So I am ok with using `EntitlementValue` here, but I'd improve on this so make the API spec more clean.

@GAlexIHU